### ISSUE TITLE
GitHub Desktop App Icon not shown

### ISSUE THREAD

**Issue by AluminiumTech** [shiftkey/desktop#16]
Issue: I don't know if this is intentional or not but the GitHub Desktop App icon is not shown. And the icon shown for when a window does open is weird looking.

The GitHub Desktop window which is open is the one with the red stop/close thing in the icon. And the one on the left of the screenshot tool is the GitHub Desktop App icon although as you can no doubt see, it is not the real GitHub Desktop icon.

![](https://user-images.githubusercontent.com/14107316/34097309-ad81932a-e413-11e7-8a50-f54772b15965.png)

GitHub Desktop Version: 1.0.11 RC1 for Linux

**OS version:** Ubuntu 17.10 with latest updates.

### Steps to Reproduce

1. Pin GitHub Desktop App to favorites/taskbar
2. Launch GitHub Desktop App

**Expected behavior:** Official GitHub Desktop app icon to be used in taskbar/favorites and to create a window with that icon.

**Actual behavior:** A generic app icon was used in taskbar/favorites and it created a window a weird icon.

**Reproduces how often:** [Every Time

**Comment by shiftkey** [comment]
@AluminiumTech can you try out [RC2](https://github.com/shiftkey/desktop/releases/tag/release-1.0.11-linux2)? I think this was fixed in #11 but I couldn't confirm it on Ubuntu 16.04 (see https://github.com/shiftkey/desktop/pull/11#discussion_r157325237 for more context).

**Comment by AluminiumTech** [comment]
@shiftkey In RC2, I see it now but it's super pixelated and also it doesn't show in the correct icon in "Show Applications."

**Comment by shiftkey** [comment]
@AluminiumTech can you attach screenshots? The source image should be 512x512...

**Comment by AluminiumTech** [comment]
Sorry for the late reply. Here's what it looks like for me.

![](https://user-images.githubusercontent.com/14107316/34158413-49708560-e500-11e7-9df6-556b91b05169.png)

**Comment by shiftkey** [comment]
Any thoughts @picandocodigo - should we be using a smaller image here?

**Comment by picandocodigo** [comment]
Maybe we should use a different 128x128?

I don't know if it would be possible with Electron to set different icon sizes for different situations, but I found [this](https://askubuntu.com/questions/141594/where-are-the-different-app-icon-sizes-submitted-through-the-app-developer-site) specific to Ubuntu:

> 128: App details view in the Software Center, Alt-tab Switcher
> 64 : Dash, and resized to 48 for the Launcher and the Software Center tiled view
> 32 : Software Center list view
> 16 : Some places on the desktop (Nautilus list view?) and probably in the software update

**Comment by shiftkey** [comment]
The VSCode icon in the Linux build (both in the repository and the shipped release) is 1024x1024, which really doesn't help here:

<img width="476" src="https://user-images.githubusercontent.com/359239/34185587-23b8787a-e576-11e7-8a2b-80f0b3b68e86.png">

**Comment by AluminiumTech** [comment]
Idk, the VSCode icon on my taskbar looks just fine.

**Comment by shiftkey** [comment]
I can see a variation of this issue in my Fedora 27 VM using the RC3 AppImage install:

 - when searching for `Desktop` the icon renders fine

<img width="439" src="https://user-images.githubusercontent.com/359239/34347779-e7f44fbe-ea55-11e7-8095-626e300237ce.png">

 - when using <kbd>ALT</kbd> + <kbd>TAB</kbd> to switch between windows, the icon does appear aliased:

<img width="378" src="https://user-images.githubusercontent.com/359239/34347788-fdb9183e-ea55-11e7-9926-2df631f0968f.png">

**Comment by appelgriebsch** [comment]
I fixed some issues on my GNOME3 desktop with GitHub Desktop RC3 by changing the content of /usr/share/applications/desktop.desktop to:
``
[Desktop Entry]
Name=GitHub Desktop
Comment=Simple collaboration from your desktop
Exec="/opt/GitHubDesktop/desktop" %U
Terminal=false
Type=Application
Icon=github
StartupWMClass=github desktop
Categories=GNOME;GTK;Development;
``

**Comment by shiftkey** [comment]
@appelgriebsch I'm intrigued by those changes:

- **Icon** should map to something inside the distribution. Here's what gets generated for AppImage:

```
$ cd dist/__appImage-x64/usr/share/icons/
$ tree .
.
└── hicolor
    ├── 1024x1024
    │   └── apps
    │       └── desktop.png
    ├── 256x256
    │   └── apps
    │       └── desktop.png
    └── 512x512
        └── apps
            └── desktop.png

7 directories, 3 files
```

So where is you `github` icon coming from?

 - I wasn't sure what **StartupWMClass** represented, but [this answer](https://askubuntu.com/a/367851) seems to suggest its a nice bit of helpful polish.

I've figured out how to add custom values to the `.desktop` file at package time, so it's possible to add this value in alongside whatever is generated...

**Comment by Cromefire** [comment]
I don't know if this works, but have you ever tried to use an svg? In this case the OS decides, wich is the best size.

**Comment by gengjiawen** [comment]
I reported an issue to `electron-builder` https://github.com/electron-userland/electron-builder/issues/2098, but got no reply for now.

**Comment by ziggy42** [comment]
Probably related:
![screenshot from 2018-01-18 18-02-17](https://user-images.githubusercontent.com/4258550/35110968-6308db30-fc7a-11e7-8d68-578c60d34380.png)

:sweat_smile:

**Comment by j-f1** [comment]
@Serkan-devel Given that the text in the image looks normal, it’s probably the icon that’s squashed.

**Comment by ziggy42** [comment]
Yes, it's the icon

**Comment by shgysk8zer0** [comment]
I've had all these issues and more. I have the `.deb` for Ubuntu 17.10 and the `.rpm` for Fedora 27.

On Ubuntu, it was at first showing the desktop icon (as in the brown folder one). I renamed `/usr/share/hicolor/*/apps/desktop.png` to be `github.png` instead and changed the `Icon` in `desktop.desktop`, and now I've got a generic icon. I even tried creating an SVG to put in `hicolor/scalable/apps/` but that had the same results. Also, realized the [SVG version](https://desktop.github.com/images/desktop-icon.svg) is the first thing you'll find on the GitHub Desktop site.

I'm also having duplicate icons on Fedora. I pinned it, but when opening it there's a second icon that shows up. Have I missed somebody mentioning that?

![](https://i.imgur.com/GdWu2HD.png)

**Comment by Huluti** [comment]
Any news?

**Comment by gabrieleara** [comment]
Hi,
I wrote a small workaround for this issue that worked for me using Ubuntu 18.04 and Gnome.
Basically it incorporates both fixes suggested by @appelgriebsch in this thread and @jkbest2 on #42.

The solution is contained in the bash script attached to this comment as a zip archive (bash script are not supported as attachment). You can get it here [githubfix.zip](https://github.com/shiftkey/desktop/files/2277634/githubfix.zip) or check the code below.

The first part changes the file of the icon in hicolor theme from `desktop.png` to `github-desktop.png` name, to avoid conflicts with already-existing system icon with that same name.
The second part changes the icon reference in a new .desktop file (that overrides default one provided with the .deb installation) and sets `StartupWMClass=desktop` flag, which tells the desktop manager to treat that app as a desktop app. This forces the usage of theme-provided icon as the icon both in app drawer and when the application is running.

``` bash
#!/bin/bash

# After installing the deb file, execute this script

sudo mv /usr/share/icons/hicolor/1024x1024/apps/desktop.png /usr/share/icons/hicolor/1024x1024/apps/github-desktop.png
sudo mv /usr/share/icons/hicolor/512x512/apps/desktop.png /usr/share/icons/hicolor/512x512/apps/github-desktop.png
sudo mv /usr/share/icons/hicolor/256x256/apps/desktop.png /usr/share/icons/hicolor/256x256/apps/github-desktop.png

cp /usr/share/applications/desktop.desktop ~/.local/share/applications/

sed -i -e 's/=desktop/=github-desktop/g' ~/.local/share/applications/desktop.desktop
echo "StartupWMClass=github desktop" >> ~/.local/share/applications/desktop.desktop
```

I'll also attach here some screenshot to show how this works in my settings.

![screen1](https://user-images.githubusercontent.com/11692280/43954561-3a15e3bc-9c9d-11e8-935a-9c2cf040968c.png)
![screen2](https://user-images.githubusercontent.com/11692280/43954560-39fae1d4-9c9d-11e8-9a93-b675f0400fcc.png)
![screen3](https://user-images.githubusercontent.com/11692280/43954559-39defeba-9c9d-11e8-9c1f-5166576c40a5.png)

**Comment by cromefire** [comment]
> `cp /usr/share/applications/desktop.desktop ~/.local/share/applications/`

@gabrieleara why does it need to be local?

**Comment by gabrieleara** [comment]
@chromefire actually it doesn’t have to be, I just found it more convenient than modifying the original file because
- I usually work on machines with only my user
- user-specific entries take precedence over global ones when concerning .desktop entries, as stated [here](https://wiki.archlinux.org/index.php/desktop_entries#Application_entry)
- I don’t know how exactly how deb installer worlds, but if after an application update the desktop file is overwritten, then the new desktop file shall be changed as well, while in this case it is not needed. Also if updating the application re-creates icons it’s not a problem since their copies would remain there.

So it’s not strictly necessary, just a personal preference that I though many users would share.

**Comment by cromefire** [comment]
Okay but that is nothing that should be included in the actual fix

**Comment by gabrieleara** [comment]
No, indeed, it’s just a workaround that could be used while this problem is not fixed.

**Comment by ThibaudDauce** [comment]
On KDE there is a Breeze icon named `/usr/share/icons/breeze/places/64/desktop.svg` which override the Github `/usr/share/icons/hicolor/1024x1024/apps/desktop.png` icon.

**Comment by shiftkey** [comment]
I believe most of these issues were related to the name of the package, which was changed from `desktop` to `github-desktop` in #80 and is available with the [`1.5.0-linux1`](https://github.com/shiftkey/desktop/releases/tag/release-1.5.0-linux1) release.

I'm going to close this out but please open a fresh issue if you notice any ongoing issues with the current installer.
